include/openssl/core_numbers.h

```diff
@@ -312,6 +313,33 @@ OSSL_CORE_MAKE_FUNC(int, OP_kdf_get_ctx_params,
 OSSL_CORE_MAKE_FUNC(int, OP_kdf_set_ctx_params,
                     (void *kctx, const OSSL_PARAM params[]))
 
+/* RAND */
+
+# define OSSL_FUNC_RAND_NEWCTX                        1
+# define OSSL_FUNC_RAND_FREECTX                       2
+# define OSSL_FUNC_RAND_GET_BYTES                     3
+# define OSSL_FUNC_RAND_RESEED                        4
+# define OSSL_FUNC_RAND_GETTABLE_PARAMS               5
+# define OSSL_FUNC_RAND_GETTABLE_CTX_PARAMS           6
+# define OSSL_FUNC_RAND_SETTABLE_CTX_PARAMS           7
+# define OSSL_FUNC_RAND_GET_PARAMS                    8
+# define OSSL_FUNC_RAND_GET_CTX_PARAMS                9
+# define OSSL_FUNC_RAND_SET_CTX_PARAMS               10
+
+OSSL_CORE_MAKE_FUNC(void *, OP_rand_newctx, (void *provctx))
+OSSL_CORE_MAKE_FUNC(void, OP_rand_freectx, (void *rctx))
+OSSL_CORE_MAKE_FUNC(int, OP_rand_get_bytes, (void *rctx, void *out,
+                                             size_t outlen))
+OSSL_CORE_MAKE_FUNC(int, OP_rand_reseed, (void *rctx))
+OSSL_CORE_MAKE_FUNC(const OSSL_PARAM *, OP_rand_gettable_params, (void))
+OSSL_CORE_MAKE_FUNC(const OSSL_PARAM *, OP_rand_gettable_ctx_params, (void))
+OSSL_CORE_MAKE_FUNC(const OSSL_PARAM *, OP_rand_settable_ctx_params, (void))
+OSSL_CORE_MAKE_FUNC(int, OP_rand_get_params, (OSSL_PARAM params[]))
+OSSL_CORE_MAKE_FUNC(int, OP_rand_get_ctx_params,
+                    (void *rctx, OSSL_PARAM params[]))
+OSSL_CORE_MAKE_FUNC(int, OP_rand_set_ctx_params,
+                    (void *rctx, const OSSL_PARAM params[]))
+
```

Comment:
`doc/man7/provider-rand.h`?

Reply:
Ah, missed this one.  The new stuff has [tentacles](http://hplhs.org/freebies/audiosamples/tentaclessample.mp3) everywhere....

Reply:
Interestingly, there is no provider-kdf documentation file :(

Reply:
Done, crudely.